### PR TITLE
Feat/fcc 14 signup route

### PR DESCRIPTION
**Pull request by Sameer55tech**
Added signup route

### CODE REVIEW

File: src/core/controller/User/userController.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    const { firstName, lastName, gender, email, password } = request.body;
+    if (!(validator.isEmail(email) && validator.isStrongPassword(password) && isLocale(firstName) && isLocale(firstName))) {
+      return sendResponse(onError(403, messageResponse.INVALID_INPUT), res);
+    }
+    const userExists = await userModel.findOne({ where: { email:email } });
+    if (userExists) {
+      return sendResponse(onError(409, messageResponse.Email_Exist), response);
+    }
+    const salt = await bcrypt.genSalt(10);
+    const hashedPass = await bcrypt.hash(password, salt);
```
Comment (sushma0055) [inline src/core/controller/User/userController.js]: hashedPassword seems better. Rename the variable

File: src/core/controller/User/userController.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    const user = await userModel.findByPk(request.params.id);
+    if (!user) {
+      return response.status(404).send('User not found');
+    }
+    const salt = await bcrypt.genSalt(10);
+    const hashedPass = await bcrypt.hash(password, salt);
+    await userModel.update({ firstName, lastName, email, password: hashedPass }, { where: { id: request.params.id }});
+    return response.status(200).send('User updated successfully');
+  } catch (error) {
+    console.log(error);
```
Comment (sushma0055) [inline src/core/controller/User/userController.js:54]: Remove console statements while commiting

File: src/core/controller/login/loginController.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      return sendResponse(onError(403, messageResponse.INVALID_INPUT), res);
+    }
+
+    const userExist = (await userModel.findOne({ where: { email } })).toJSON();
+    
+    if (userExist) {
+      const checker =  bcrypt.compareSync( password,userExist["password"]);
+
+      if (checker) {
+        var token = jwt.sign({ email, password }, config.SECRET, {
```
Comment (sushma0055) [inline src/core/controller/login/loginController.js]: avoid usage of var. use let / const

File: src/core/helper/constants.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const LOGIN_SUCCESSFULLY = "Successfully logged in";
+const INVALID_INPUT='Invalid input data';
+const INVALID_PASSWORD='Invalid password entered';
+const USER_NOT_EXIST='user does not exist';
+const Error='Error while fetching data'
```
Comment (sushma0055) [inline src/core/helper/constants.js]: maintain consistency. All variables should be Capitalized if constants.

File: src/core/helper/responses.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+function onError(statusCode, message) {
+  return {
+    statusCode,
+    message,
+    type: "failure",
+  };
+}
+function sendResponse(resObject, res) {
+  if (resObject.type === "success") {
+    // console.log("-----------------------------------",resObject);
```
Comment (sushma0055) [inline src/core/helper/responses.js]: remove this

File: src/server/server.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+//     definition: {
+//       openapi: "3.0.0",
+//       info: {
+//         title: "55 Chit Chat User API",
+//         version: "1.0.0",
+//         description:
+//           "This is the user signup API for 55 Chit Chat",
+//         contact: {
+//           name: "Sameer",
+//           email: "sameer.srivastava@fiftyfivetech.io.com",
```
Comment (sushma0055) [inline src/server/server.js]: remove codes if not needed

File: src/core/controller/User/userController.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  onError,
+  sendResponse,
+  globalCatch,
+  messageResponse
+} from "../../helper";
+
+const createUser = async (request, response) => {
+  try {
+    const { firstName, lastName, gender, email, password } = request.body;
+    if (!(validator.isEmail(email) && validator.isStrongPassword(password) && isLocale(firstName) && isLocale(firstName))) {
```
Comment (tarunjain55) [inline src/core/controller/User/userController.js]: Make a validator function rather then writing here many condition. just return boolean value from that function

File: src/core/controller/User/userController.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    return response.status(200).send('User updated successfully');
+  } catch (error) {
+    console.log(error);
+    response.status(500).send({ error: error.message });
+  }
+}
+
+const deleteUser = async (request, response) => {
+  try {
+    userModel.destroy({ where: { id: request.params.id } })
```
Comment (tarunjain55) [inline src/core/controller/User/userController.js]: Make consistency in whole repo just use await instead of .then.

Comment (tarunjain55) [reply src/core/controller/User/userController.js]: remove then from every where you used

File: src/core/controller/login/loginController.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+        return sendResponse(
+          onSuccess(200, messageResponse.LOGIN_SUCCESSFULLY, token),
+
+          res
+        );
+      } else {
+        return sendResponse(
+          onError(401, messageResponse.INVALID_PASSWORD),
+
```
Comment (tarunjain55) [inline src/core/controller/login/loginController.js]: 1. remove extra space.
2. what do you mean by "res".

File: src/core/controller/login/loginController.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    const userExist = (await userModel.findOne({ where: { email } })).toJSON();
+    
+    if (userExist) {
+      const checker =  bcrypt.compareSync( password,userExist["password"]);
+
+      if (checker) {
+        var token = jwt.sign({ email, password }, config.SECRET, {
+          expiresIn: 86400,
```
Comment (tarunjain55) [inline src/core/controller/login/loginController.js]: 86400 make it env config

File: src/core/database/database.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import { Sequelize } from "sequelize";
+import logger from "../helper/logger.js";
+const sequelize = new Sequelize("ChitChat", "postgres", "16122000", {
```
Comment (tarunjain55) [inline src/core/database/database.js]: try to use env instead of using here direct.

File: src/core/helper/globalCatch.js [hunk 12]
```diff
@@ -0,0 +1,8 @@
+
+ function globalCatch(req, err) {
```
Comment (tarunjain55) [inline src/core/helper/globalCatch.js]: Don't use function keyword make it arrow function

File: src/core/helper/globalCatch.js [hunk 13]
```diff
@@ -0,0 +1,8 @@
...
+    console.log(
+      `Error in catch with route ${req.path} with method ${req.method} and with message ${err.message}`
+    );
+  }
+  export{globalCatch}
```
Comment (tarunjain55) [inline src/core/helper/globalCatch.js]: use prettier. it bad practice to push code without format

File: src/core/helper/responses.js [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    const result = {
+      status: resObject.type,
```
Comment (tarunjain55) [inline src/core/helper/responses.js]: what do you mean by resobject

File: src/core/middleware/middleware.js [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+import config from '../../../config/config.js'
+export default {
+  // loginValidator(req, res, next) {
+  //   const { email, password } = req.body;
+  //   if (!(validator.isEmail(email) && validator.isStrongPassword(password))) {
+  //     return res
+  //       .status(401)
+  //       .send(new helper.genResponse(false, "", "unauthorized request"));
+  //   }
+  //   next();
```
Comment (tarunjain55) [inline src/core/middleware/middleware.js]: remove comment code

File: src/core/models/user/userModel.js [hunk 16]
```diff
@@ -0,0 +1,49 @@
+import { DataTypes } from "sequelize";
+import sequelize from "../../database/database.js";
+
+const User = sequelize.define("User", {
```
Comment (tarunjain55) [inline src/core/models/user/userModel.js]: table name should be in lowercase and no _ , - , and special character in it. This is not best practices for databases

File: src/core/database/database.js [hunk 17]
```diff
@@ -0,0 +1,24 @@
+import { Sequelize } from "sequelize";
+import logger from "../helper/logger.js";
+import dataMessage from "../../../config/config.js";
+
+const sequelize = new Sequelize(
+  dataMessage.DATABASE_NAME,
```
Comment (tarunjain55) [inline src/core/database/database.js]: rename it what does you mean by datamessage

**Review by tarunjain55 — APPROVED** [review]

**Pull request merged**